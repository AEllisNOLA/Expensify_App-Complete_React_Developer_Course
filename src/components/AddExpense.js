import React from 'react';

const AddExpense = () => (
    <div>
        <p>AddExpense Component!</p>
    </div>
)

export default AddExpense;