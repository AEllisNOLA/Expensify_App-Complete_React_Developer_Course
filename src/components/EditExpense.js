import React from 'react';

const EditExpense = () => (
    <div>
        <p>EditExpense Component</p>
    </div>
)

export default EditExpense;