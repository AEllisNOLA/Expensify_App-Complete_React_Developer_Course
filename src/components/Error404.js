import React from 'react';
import { Link } from 'react-router-dom';

const Error404 = () => (
    <div>
        <p>Error404 Component</p>
        <Link to='/'>Go to Dashboard</Link>
    </div>
)

export default Error404