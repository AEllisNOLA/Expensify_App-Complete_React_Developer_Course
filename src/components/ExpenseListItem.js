import React from 'react'
import { connect } from 'react-redux'
import { removeExpense } from '../actions/expenses'
import { defaultCipherList } from 'constants'

const ExpenseListItem = ({ description, amount, createdAt, dispatch, id }) => (
    <div>
    
        <h3>Description: {description}</h3>
        <p>Amount: {amount} - CreatedAt: {createdAt}</p>
        <button onClick={(e) => {
            dispatch(removeExpense({id}))

            

            
          
        }}>Remove</button>
    </div>
)


export default connect()(ExpenseListItem)


// import action generator
// connect component to access dispatch
// wire up the onclick
// use id
// you don't have to mapStateToProps
