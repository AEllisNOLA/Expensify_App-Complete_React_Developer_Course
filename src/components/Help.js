import React from 'react';

const Help = () => (
    <div>
        <p>Help Component</p>
    </div>
)

export default Help;