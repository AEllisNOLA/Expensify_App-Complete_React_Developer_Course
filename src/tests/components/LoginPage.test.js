import React from 'react'
import { shallow } from 'enzyme'
import { LoginPage } from '../../components/LoginPage'

test("Should properly render LoginPage component", () => {
    const wrapper = shallow(<LoginPage />)
    expect(wrapper).toMatchSnapshot()
})