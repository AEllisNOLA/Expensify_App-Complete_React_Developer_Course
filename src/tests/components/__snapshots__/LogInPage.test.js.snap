// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should properly render LoginPage component 1`] = `
<div>
  <button>
    Log In
  </button>
</div>
`;
